### ISSUE TITLE
ICE when invoking a modifier overriden by an unimplemented one

### ISSUE THREAD

**Issue by cameel** [ethereum/solidity#11468]
It is currently possible to override a modifier with one that has no implementation. Doing this with functions is forbidden. An attempt to invoke such a modifier results in an ICE:

```solidity
contract B {
    modifier m() virtual { _; }
}
abstract contract C is B {
    modifier m() virtual override;
}
contract D is C {
    function f() m public {}
}
```
```
Internal compiler error during compilation:
/solidity/libsolidity/ast/AST.h(1054): Throw in function const solidity::frontend::Block& solidity::frontend::ModifierDefinition::body() const
Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
std::exception::what:
[solidity::util::tag_comment*] =
```

I think this should be forbidden like it is for functions. Even then, however, it will still be possible to define an unimplemented modifier by sandwiching it between two implemented ones (see #11445) and then invoke it by qualifying it with the name of the contract where it has no implementation - this should probably be forbidden too.

## Environment
- Compiler version: 0.8.4
- Operating system: Arch Linux

**Comment by chriseth** [comment]
Agree that both (overriding and calling) should be forbidden in the same way it is for functions.

**Comment by cameel** [comment]
The issue was fixed in #11471. There was a new ICE discovered but we have a new issue for it (#11506) so I'm closing this on after all.
